adapters/avail/src/service.rs

```diff
@@ -0,0 +1,214 @@
+use core::time::Duration;
+
+use anyhow::anyhow;
+use async_trait::async_trait;
+use avail_subxt::api::runtime_types::sp_core::bounded::bounded_vec::BoundedVec;
+use avail_subxt::primitives::AvailExtrinsicParams;
+use avail_subxt::{api, AvailConfig};
+use reqwest::StatusCode;
+use sov_rollup_interface::da::DaSpec;
+use sov_rollup_interface::services::da::DaService;
+use sp_core::crypto::Pair as PairTrait;
+use sp_keyring::sr25519::sr25519::Pair;
+use subxt::tx::PairSigner;
+use subxt::OnlineClient;
+use tracing::info;
+
+use crate::avail::{Confidence, ExtrinsicsData};
+use crate::spec::block::AvailBlock;
+use crate::spec::header::AvailHeader;
+use crate::spec::transaction::AvailBlobTransaction;
+use crate::spec::DaLayerSpec;
+use crate::verifier::Verifier;
+
+/// Runtime configuration for the DA service
+#[derive(Clone, PartialEq, serde::Deserialize, serde::Serialize)]
+pub struct DaServiceConfig {
+    pub light_client_url: String,
+    pub node_client_url: String,
+    //TODO: Safer strategy to load seed so it is not accidentally revealed.
+    pub seed: String,
+}
+
+#[derive(Clone)]
+pub struct DaProvider {
+    pub node_client: OnlineClient<AvailConfig>,
+    pub light_client_url: String,
+    signer: PairSigner<AvailConfig, Pair>,
+}
+
+impl DaProvider {
+    fn appdata_url(&self, block_num: u64) -> String {
+        let light_client_url = self.light_client_url.clone();
+        format!("{light_client_url}/v1/appdata/{block_num}")
+    }
+
+    fn confidence_url(&self, block_num: u64) -> String {
+        let light_client_url = self.light_client_url.clone();
+        format!("{light_client_url}/v1/confidence/{block_num}")
+    }
+
+    pub async fn new(config: DaServiceConfig) -> Self {
+        let pair = Pair::from_string_with_seed(&config.seed, None).unwrap();
+        let signer = PairSigner::<AvailConfig, Pair>::new(pair.0.clone());
+
+        let node_client = avail_subxt::build_client(config.node_client_url.to_string(), false)
+            .await
+            .unwrap();
+        let light_client_url = config.light_client_url;
+
+        DaProvider {
+            node_client,
+            light_client_url,
+            signer,
+        }
+    }
+}
+
+const POLLING_TIMEOUT: Duration = Duration::from_secs(60);
+const POLLING_INTERVAL: Duration = Duration::from_secs(1);
```

Comment:
Are these values supposed to be the same all the time? 

Because other option would be to put them in DaServiceConfig.

Reply:
Fixed by #980